core/src/main/scala/org/apache/spark/deploy/SparkCuratorUtil.scala

```diff
@@ -35,8 +35,11 @@ private[spark] object SparkCuratorUtil extends Logging {
   def newClient(
       conf: SparkConf,
       zkUrlConf: String = "spark.deploy.zookeeper.url"): CuratorFramework = {
```

Comment:
You meant the param zkUrlConf? It feels like a good abstraction so that the url configuration can at least be overriden right?